src/Symfony/Component/Console/EventListener/ExceptionListener.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Symfony\Component\Console\EventListener;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\ConsoleEvents;
+use Symfony\Component\Console\Event\ConsoleExceptionEvent;
+use Symfony\Component\Console\Event\ConsoleTerminateEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Console exception listener.
+ *
+ * Attempts to log exceptions or abnormal terminations of console commands.
+ *
+ * @author  James Halsall <james.t.halsall@googlemail.com>
+ */
+class ExceptionListener implements EventSubscriberInterface
+{
+    protected $logger;
+
+    /**
+     * Constructor.
+     *
+     * @param LoggerInterface $logger A logger
+     */
+    public function __construct(LoggerInterface $logger = null)
+    {
+        $this->logger = $logger;
+    }
+
+    /**
+     * Handles console command exception.
+     *
+     * @param ConsoleExceptionEvent $event Console event
+     */
+    public function onKernelException(ConsoleExceptionEvent $event)
+    {
+        if (null === $this->logger) {
+            return;
+        }
+
+        $exception = $event->getException();
+
+        $this->logger->error($exception->getMessage(), array('exception' => $exception));
+    }
+
+    /**
+     * Handles termination of console command.
+     *
+     * @param ConsoleTerminateEvent $event Console event
+     */
+    public function onKernelTerminate(ConsoleTerminateEvent $event)
+    {
+        if (null === $this->logger) {
+            return;
+        }
+
+        $exitCode = $event->getExitCode();
+
+        if ($exitCode === 0) {
+            return;
+        }
+
+        if ($exitCode > 255) {
+            $event->setExitCode(255);
```

Comment:
IMHO, this should not be done here.
And BTW, it's already done there https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Console/Application.php#L141-L147